### ISSUE TITLE
Dubious error message when indexing an array with a substraction expression

### ISSUE THREAD

**Issue by asahi3g** [skycoin/cx#104]
```
package main

var array [8]i32
var count i32 = 4

func main() {
    var i i32 = array[count - 1]
}
```
output:
```
error: pending_8.cx:8 identifier '*lcl_0' does not exist
```

**Comment by corpusc** [comment]
this is probably irrelevant to you, but
in case you are trying to get something done (in additon to documenting errors)...

...shorthand operations (in general) are very iffy.  
i32.sub(count, 1) seems reliable.

**Comment by asahi3g** [comment]
Thanks for writing workarounds, I'll try to add them when opening the issues. Feel free to add some more.

**Comment by amherag** [comment]
Fixed in #228
